src/main/java/az/gdg/msauth/config/CacheConfig.java

```diff
@@ -0,0 +1,35 @@
+package az.gdg.msauth.config;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.cache.CacheManager;
+import org.springframework.cache.annotation.EnableCaching;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.scheduling.annotation.EnableScheduling;
+import org.springframework.scheduling.annotation.Scheduled;
+
+import java.time.LocalDateTime;
+
+@Configuration
+@EnableCaching
+@EnableScheduling
+public class CacheConfig {
+
+    private static final Logger logger = LoggerFactory.getLogger(CacheConfig.class);
+    private final CacheManager cacheManager;
+
+    public CacheConfig(CacheManager cacheManager) {
+        this.cacheManager = cacheManager;
+    }
+
+    @Scheduled(cron = "0 59 23 * * ?")  // at 23:59 every day
+    public void clearCacheSchedule() {
+        logger.info("Config.clearCacheSchedule.start : date {}", LocalDateTime.now());
+        for (String name : cacheManager.getCacheNames()) {
+            cacheManager.getCache(name).clear();
+        }
+
+        logger.info("Config.clearCacheSchedule.stop.success : date {}", LocalDateTime.now());
```

Comment:
No need to write the date. It is already in each log, for example : 
May 14 11:50:27 gdg-ms-auth app/web.1 2020-05-14 11:50:27.132  INFO 4 --- [           main] o.s.i.endpoint.EventDrivenConsumer       : started bean '_org.springframework.integration.errorLogger'
Look it up in papertrail, please